interp/src/flatten/structures/environment/program_counter.rs

```diff
@@ -56,6 +56,67 @@ impl ControlPoint {
             false
         }
     }
+
+    /// Returns a string showing the path from the root node to input node.
+    /// How to get context?
+    pub fn string_path(&self, ctx: &Context) -> String {
+        // Does it matter if it takes ownership?
+        let path = SearchPath::find_path_from_root(self.control_node_idx, ctx);
+        let control_map = &ctx.primary.control;
+        let mut string_path = String::from("");
+        let mut count = -1;
+        let mut body = false;
+        let mut if_branches: HashMap<ControlIdx, String> = HashMap::new();
```

Comment:
I think this approach is fine and there's no need to change it here, but wanted
to flag the fact that if you `zip` the path with itself staggered by one
element, you can look at both the current node and the parent without needing
auxiliary structures. This can either be accomplished by processing the first
node specially or by appending a `None` value and mapping the parent field to an
option